From 0671a16d727608ef683f3f98691ac027b500ec46 Mon Sep 17 00:00:00 2001
From: Fabrice Goucem <fabrice.goucem@nxp.com>
Date: Mon, 25 Oct 2021 17:18:31 +0200
Subject: [PATCH 01/27] arm64: dts: imx8mm-evk-harpoon, imx8mn-evk-harpoon,
 imx8mp-evk-harpoon: new DTBs for real-time-edge

Signed-off-by: Fabrice Goucem <fabrice.goucem@nxp.com>
[replace spaces by tabulations]
Signed-off-by: Stephane Viau <stephane.viau@nxp.com>
(rebase on lf-5-15.y @commit 29608d7e7544908730ab43e1a65056e345f5c1ae)
Signed-off-by: Marouen Ghodhbane <marouen.ghodhbane@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile            |  5 +++--
 .../boot/dts/freescale/imx8mm-evk-harpoon.dts     | 15 +++++++++++++++
 .../boot/dts/freescale/imx8mn-evk-harpoon.dts     | 15 +++++++++++++++
 .../boot/dts/freescale/imx8mp-evk-harpoon.dts     | 15 +++++++++++++++
 4 files changed, 48 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mn-evk-harpoon.dts
 create mode 100644 arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index b919d438d9c4..9b6a1d85f51a 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -85,6 +85,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mm-evk.dtb imx8mm-evk-rpmsg.dtb imx8mm-evk-rm67191
 			  imx8mm-evk-usd-wifi.dtb \
 			  imx8mm-evk-qca-wifi.dtb \
 			  imx8mm-evk-dpdk.dtb \
+			  imx8mm-evk-harpoon.dtb \
 			  imx8mm-evk-rm67199.dtb imx8mm-evk-rm67191-cmd-ram.dtb imx8mm-evk-rm67199-cmd-ram.dtb \
 			  imx8mm-evk-lk.dtb imx8mm-evk-rpmsg-wm8524-lpv.dtb \
 			  imx8mm-evk-rpmsg-8m-buf.dtb \
@@ -132,7 +133,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mn-ddr4-evk.dtb imx8mn-ddr4-evk-rm67191.dtb imx8mn
 			  imx8mn-ddr4-evk-usd-wifi.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mn-ddr4-evk-rpmsg.dtb imx8mn-ddr3l-evk-rpmsg.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mn-ddr4-evk-root.dtb imx8mn-ddr4-evk-inmate.dtb imx8mn-evk-root.dtb imx8mn-evk-inmate.dtb \
-			  imx8mn-evk-lk.dtb imx8mn-ddr4-evk-lk.dtb
+			  imx8mn-evk-lk.dtb imx8mn-ddr4-evk-lk.dtb imx8mn-evk-harpoon.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mn-ddr3l-evk-ak5558.dtb imx8mn-ddr4-evk-ak5558.dtb imx8mn-evk-ak5558.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mn-ab2.dtb imx8mn-ddr3l-ab2.dtb imx8mn-ddr4-ab2.dtb
 dtb-$(CONFIG_ARCH_MXC) += imx8mn-evk-iqaudio-dacplus.dtb imx8mn-evk-iqaudio-dacpro.dtb imx8mn-evk-hifiberry-dacplus.dtb \
@@ -140,7 +141,7 @@ dtb-$(CONFIG_ARCH_MXC) += imx8mn-evk-iqaudio-dacplus.dtb imx8mn-evk-iqaudio-dacp
 dtb-$(CONFIG_ARCH_MXC) += imx8mp-evk.dtb imx8mp-evk-rm67191.dtb imx8mp-evk-it6263-lvds-dual-channel.dtb \
 			  imx8mp-evk-pcie-ep.dtb  imx8mp-evk-rpmsg.dtb imx8mp-evk-ecspi-slave.dtb \
 			  imx8mp-evk-jdi-wuxga-lvds-panel.dtb imx8mp-evk-flexcan2.dtb \
-			  imx8mp-evk-root.dtb imx8mp-evk-inmate.dtb imx8mp-evk-ov2775.dtb \
+			  imx8mp-evk-root.dtb imx8mp-evk-harpoon.dtb imx8mp-evk-inmate.dtb imx8mp-evk-ov2775.dtb \
 			  imx8mp-evk-ov2775-ov5640.dtb imx8mp-evk-basler-ov5640.dtb imx8mp-evk-basler.dtb \
 			  imx8mp-evk-basler-ov2775.dtb imx8mp-evk-dual-basler.dtb \
 			  imx8mp-evk-dual-ov2775.dtb imx8mp-evk-spdif-lb.dtb imx8mp-evk-dsp.dtb \
diff --git a/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts b/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
new file mode 100644
index 000000000000..47e16e606332
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mm-evk-harpoon.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2021 NXP
+ */
+
+#include "imx8mm-evk-root.dts"
+
+&i2c3 {
+       status = "disabled";
+};
+
+&iomuxc {
+       pinctrl-names = "default";
+       pinctrl-0 = <&pinctrl_i2c3>;
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mn-evk-harpoon.dts b/arch/arm64/boot/dts/freescale/imx8mn-evk-harpoon.dts
new file mode 100644
index 000000000000..2700440df445
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mn-evk-harpoon.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2021 NXP
+ */
+
+#include "imx8mn-evk-root.dts"
+
+&i2c3 {
+       status = "disabled";
+};
+
+&iomuxc {
+       pinctrl-names = "default";
+       pinctrl-0 = <&pinctrl_i2c3>;
+};
diff --git a/arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts b/arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts
new file mode 100644
index 000000000000..cc153b97339e
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx8mp-evk-harpoon.dts
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2021 NXP
+ */
+
+#include "imx8mp-evk-root.dts"
+
+&i2c3 {
+	status = "disabled";
+};
+
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3>;
+};
-- 
2.34.1

